src/AELFFaucet.Application/Project/ClaimService.cs

```diff
@@ -106,17 +163,61 @@ public async Task<MessageResult> ClaimTokenAsync(string walletAddress)
     }
 
     [HttpPost("api/claim-seed")]
-    public async Task<MessageResult> ClaimNFTSeedAsync(string walletAddress)
+    public async Task<MessageResult> ClaimNFTSeedAsync(string walletAddress, string recaptchaToken, [FromHeader(Name = "Platform")] string platform)
     {
+        var messageResult = new MessageResult();
+
+        // Verify reCAPTCHA token here based on platform
+        var secretKey = GetRecaptchaSecretKey(platform);
+        if (string.IsNullOrEmpty(secretKey))
+        {
+            messageResult.IsSuccess = false;
+            messageResult.Code = Convert.ToInt32(CodeStatus.InvalidPlatform);
+            messageResult.Message = "Invalid platform specified.";
+            return messageResult;
+        }
+        // Verify reCAPTCHA token here
+        var isCaptchaValid = await VerifyRecaptchaAsync(recaptchaToken, secretKey);
+
+        if (!isCaptchaValid)
+        {
+            messageResult.IsSuccess = false;
+            messageResult.Code = Convert.ToInt32(CodeStatus.InvalidCaptcha);
+            messageResult.Message = "Invalid reCAPTCHA verification.";
+            return messageResult;
+        }
+
         return await ClaimSeedAsync(walletAddress, _otherTokenSendContractService, tokenInfo => tokenInfo.IsSentSeed, (tokenInfo, isSent) => tokenInfo.IsSentSeed = isSent);
     }
 
     [HttpPost("api/claim-nft-seed")]
-    public async Task<MessageResult> ClaimSeedAsync(string walletAddress)
+    public async Task<MessageResult> ClaimSeedAsync(string walletAddress, string recaptchaToken, [FromHeader(Name = "Platform")] string platform)
```

Comment:
You need to add this new function signature to IClaimService. The same for the rest where you changed function definition.